### ISSUE TITLE
Website migration 

### ISSUE THREAD

**Issue by gfenoy** [ZOO-Project/migration#7]
The ZOO-Project.org website relies on Trac Wiki pages.

We should probably think of a new way to handle our web site. Indeed, I was proposing in #5 to keep a read-only Trac. But, if we continue relying on Trac Wiki pages, it means that to update a page, you will have to modify the Trac Wiki page. In consequence, the Trac cannot be set read-only.

I feel we should find another way to manage the ZOO-Project.org website, preferably using GitHub.

**Comment by jmckenna** [comment]
I strongly agree that we must move away from relying on Trac Wiki pages, for the official homepage of the project.

**Comment by gfenoy** [comment]
We have different kind of page generated from the ZOO-Project.org.
* Pure HTML content, for instance: https://svn.zoo-project.org/trac/wiki/ZooWebSite/2015/ZOO-Project/ZOO-Identity
* Combined Wiki Page using the Trac Include macro, i.e. https://svn.zoo-project.org/trac/wiki/ZooWebSite/2015/Code/Download
* wiki page combining HTML and Wiki syntaxe, i.e. https://svn.zoo-project.org/trac/wiki/Downloads2

The WPS service Identifier used to print web pages out of xmlrpc read from Trac is available in [wdisplay.py](https://github.com/ZOO-Project/migration/blob/main/wservice.py).

**Comment by gfenoy** [comment]
I did try to use [fosshost](https://fosshost.org/) but unfortunately, I received an email over the week-end asking for support from our sponsors. Not to say that it is not sustainable but I wonder as they were asking already for support if it is a really sustainable solution.

I think it will be difficult to ask for more money from our sponsors for the infrastructure required for the ZOO-Project web-site. For 12 years now, both [Gateway Geomatics](https://gatewaygeomatics.com/) and [GeoLabs](http://www.geolabs.fr/) are paying and offer their time to have the infrastructure up and running most of the time. So, I wonder if it is the right moment to migrate our entire infrastructure to a tier.

I would like to know what do others think?

CC @ZOO-Project/psc

**Comment by gfenoy** [comment]
Just a short update, here is the list of pages to be migrated:

 * [Home page](http://zoo-project.org/)
 * About
   * [Tribe](http://zoo-project.org/ZOO-Project/ZOO-Tribe)
   * [Events](http://zoo-project.org/ZOO-Project/ZOO-Events)
   * [Identity](http://zoo-project.org/ZOO-Project/ZOO-Identity)
 * Ressources:
   * [Download](http://zoo-project.org/new/Code/Download) 
   * [Documentation (landing page))](http://zoo-project.org/new/Resources/Documentation)
   * [Workshops (landing page)](http://zoo-project.org/new/Resources/Workshops)
   * [Examples](http://zoo-project.org/Resources/Examples)
   * [Presentations](http://zoo-project.org/Resources/Presentations)
* Get involved:
   * [Sponsorship](http://zoo-project.org/new/Get%20involved/Sponsorship)
   * [Knowledge partners](http://zoo-project.org/new/Get%20involved/Knowledge%20partners)
   * [Feed the animals](http://zoo-project.org/new/Get%20involved/Feed%20the%20animals)

So 12 pages should be migrated to GitHub pages.

For the rest, including workshops and presentations slides, they already are stored using GitHub pages. Still, they should probably be dynamically produced using a dedicated GitHub action.

**Comment by gfenoy** [comment]
I wonder if we should not simply use a docker image dedicated to building the web site from the old Trac wiki pages as it is now then deploy the result as GitHub pages...

**Comment by venka-foss4g** [comment]
+1 zoo-project.org landing page could point to appropriate page on git-hub and all the menu, links on zoo-project.org could also incorporated on this page.

**Comment by gfenoy** [comment]
Maybe using a free tier on Microsoft Azure can be a solution for hosting the demonstrations part of the web site.

More information [here](https://cloudblogs.microsoft.com/opensource/2021/09/28/announcing-azure-credits-for-open-source-projects/) and application form is here: https://opensource.microsoft.com/azure-credits#credits-apply

Deploying the containers setup using docker-compose should be enough to ensure the demonstration. availability.
